### ISSUE TITLE
Question on how to implement this

### ISSUE THREAD

**Issue by Snidd** [joheredi/openjs-world-2021#2]
Hi,

Awesome typing, i'm trying to wrap my head around how to implement this with fetch, and adding headers to the calls. 

But i can't figure out how to use the typing in an actual method that does the calls, could you give an example or point me in the right direction? :)

**Comment by bprager** [comment]
Same question. It looks like great concept.
How do I implement something using the BookLibrary interface? A little example would be helpful. Thank you.

**Comment by cainthebest** [comment]
This is true, an example with node fetch or axios would be great as i have already spent a few hours on trying to implement this. I would've thought that you use it with a class and constructor but ran into a few problems and when I found a solution it was too far from the concept.

**Comment by pablo-rufat-secture** [comment]
Anything on this?

**Comment by cainthebest** [comment]
I based a api wrapper on it kinda
https://github.com/cainthebest/nitrado.js

**Comment by pablo-rufat-secture** [comment]
Thank you so much!

**Comment by claudiomed** [comment]
Does anybody have an example on how to implement this? @cainthebest 's repo looks like has been deleted and I am trying to figure out how this is implemented with Axios, for instance.
